syntax = "proto3";

package users;

message User {
    string uuid = 1;
    string username = 2;
    string password = 3;
    string email = 4;
    string first_name = 5;
    string last_name = 6;
    bool   is_admin = 7;
    bool   is_active = 8;
    int64  last_login = 9;
    int64  date_joined = 10;
}


message RetrieveUserByIDRequest {
    string uuid = 1;
}

service Users {
    rpc GetUserByID (RetrieveUserByIDRequest) returns (User);
}
